### ISSUE TITLE
v055d: DNS-only mode broken?

### ISSUE THREAD

**Issue by ignoramous** [celzero/rethink-app#1330]
>  Technical details about my setup : I am on android 8.1 (stock) which does not support setting up custom DNS natively. I configured Rethink app with on-device blocklist with marking some domains/websites as "trusted" in Firewall-IP&Ports rules (using RethinkDNS Plus), mode set as DNS and Firewall (default) mode, blocked internet access for many apps in firewall, IP version as Auto. I was using app (with configuration mentioned above) in my main android user profile since release of v055c with no issues at all (except battery usage & IPv6 showing up as "unknown"in Network log UI). I am using mobile data and no router or wifi.
>
>I will open new issue just for battery issue.
>
>My assumption is, the issue is not caused by any blocklists or firewall as everything was working well in v055c.
>
>I shared logs and relevant screenshots with @ignoramous  by email.
>
>So, everything was working fine in v055c. I updated to v055d and noticed some apps couldn't get connected while others worked fine, some websites not working while others were working fine. There was no sign of blocking any app or website in Network log UI by either firewall or DNS. If I switched to netguard or no firewall at all, everything seemed to working as expected.
>
> I also tried to make startpage.com as trusted domain, set Session app to bypass firewall and DNS, but that didn't seem to have any difference.
>
>I could reproduce this issue on different device with fresh install of rethink-app (with all by-default configuration).
>
>Steps to reproduce the issue :
> 1. Set only DNS mode for rethink app. Keep all other configuration as default.
> 2. Click on configure > Networks > Choose IP Version and set it to IPv6.
>3. Open Browser.
>4. Go to website startpage.com and observe.
>5. Go to website wikipedia.org and observe.
>6. Open Session messenger and observe.
>
>I force closed browser(s), Rethink app and session messenger after trying each case and restarted them for next case. My results for following cases, note that wikipedia is working in all below cases :
>
>1. Only firewall mode + network set to IPv6 = Startpage works fine. But session app doesn't work (I am not sure if it is showing up into Network logs UI, as everything is marked as "unknown", but nothing seems to be blocked by firewall).
>
>2. Only DNS mode + network set to IPv6 = Startpage doesn't work, shows unreachable or address not found error (tried more than one browser to make sure it is not browser specific issue). But session app doesn't work (session doesn't appear in Network logs UI at all in this case, My guess is, it is because session uses their own dns or i2p? not sure.).
>
>3. Only firewall mode + network set to IPv4 = Startpage works fine. Session also works fine and shows up in Netwok logs UI.
>
>4. Only DNS mode + network set to IPv4 = Startpage works fine. Session also works fine.

_Originally posted by @Sinestr0 in https://github.com/celzero/rethink-app/issues/1325#issuecomment-2033655957_

**Comment by ignoramous** [comment]
Rethink doesn't restart the tunnel on mode updates (*DNS-only* to *Firewall-only* to *DNS + Firewall*) and this causes all sorts of weird issues when *Advanced DNS filtering* is also turned ON along with *Choose IP version* set to *Auto*.

**Comment by hussainmohd-a** [comment]
Fix: https://github.com/celzero/rethink-app/pull/1329/commits/308271eb06053947c6d51bdd45cae28f7ce062ee
